doc/api/sqlite.md

```diff
@@ -0,0 +1,316 @@
+# SQLite
+
+<!--introduced_in=REPLACEME-->
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active development
+
+<!-- source_link=lib/sqlite.js -->
+
+The `node:sqlite` module facilitates working with SQLite databases.
+To access it:
+
+```mjs
+import sqlite from 'node:sqlite';
+```
+
+```cjs
+const sqlite = require('node:sqlite');
+```
+
+This module is only available under the `node:` scheme. The following will not
+work:
+
+```mjs
+import sqlite from 'sqlite';
+```
+
+```cjs
+const sqlite = require('sqlite');
+```
+
+The following example shows the basic usage of the `node:sqlite` module to open
+an in-memory database, write data to the database, and then read the data back.
+
+```mjs
+import { SQLiteDatabaseSync } from 'node:sqlite';
+const database = new SQLiteDatabaseSync(':memory:');
```

Comment:
That name is... unfortunate. I don't want to block anything on bikeshedding so I'm fine with this if others are, but something simple like `Database` would be fine for me.

Reply:
I was worried people would complain that "Database" is too generic in the context of a larger project like Node. I'm fine with dropping the "SQLite" prefix from the types.

The "Sync" part was to follow the fs module. SQLite is almost entirely synchronous. I thought we would probably want to expose an async API via a `node:sqlite/promises` in the future. Unfortunately, an async API can't really exist on the same database connection as the sync APIs unless we plan to make every sync access lock the entire database.

Reply:
This can also just be a function and be `sqlite.create(dbName)` (or createSync if you feel strongly about the naming/api)

Reply:
If you import from 'node:sqlite' then the `SQLite` prefix is redundant. If you import the async version from 'node:sqlite/promises' then the `Sync` postfix is redundant.